### PR TITLE
Number of partitions check

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/reference/sys/check/checks/NumberOfPartitionsSysCheck.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Override
+    public boolean validate() {
+        return validateDocTablesPartitioning(extractDocSchemaInfo());
+    }
+
+    private DocSchemaInfo extractDocSchemaInfo() {
+        DocSchemaInfo docSchemaInfo = null;
+        for (SchemaInfo schemaInfo : referenceInfos) {
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/reference/sys/check/checks/NumberOfPartitionsSysCheck.java]: implement a getSchemaInfo(String schemaName) from getSchemaInfo(TableIdent ident)  in ReferenceInfos and use that

File: sql/src/main/java/io/crate/operation/reference/sys/check/checks/NumberOfPartitionsSysCheck.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        this.referenceInfos = referenceInfos;
+    }
+
+    @Override
+    public boolean validate() {
+        return validateDocTablesPartitioning(
+                (DocSchemaInfo) referenceInfos.getSchemaInfo(DocSchemaInfo.NAME));
+    }
+
+    protected boolean validateDocTablesPartitioning(DocSchemaInfo docSchemaInfo) {
```
Comment (dobe) [inline sql/src/main/java/io/crate/operation/reference/sys/check/checks/NumberOfPartitionsSysCheck.java]: make package accessible only

**Comment by dobe** [comment]
ack

**Comment by mfussenegger** [comment]
This doesn't seem to work for custom schemas? And I think it would be great to include some pointers for users on what they can do if they have too many partitions

**Pull request merged**